src/components/modal/__internal__/modal-manager.js

```diff
@@ -1,26 +1,54 @@
 class ModalManagerInstance {
   #modalList = [];
 
-  addModal = (modal) => {
-    this.#modalList.push(modal);
+  #getTopModal() {
+    if (!this.#modalList.length) {
+      return {};
+    }
+    return this.#modalList[this.#modalList.length - 1];
+  }
+
+  addModal = (modal, setTriggerRefocusFlag) => {
+    const {
+      modal: topModal,
+      setTriggerRefocusFlag: setTrapFlag,
+    } = this.#getTopModal();
+
+    if (topModal && setTrapFlag) {
+      setTrapFlag(false);
+    }
+
+    this.#modalList.push({ modal, setTriggerRefocusFlag });
   };
 
   isTopmost(modal) {
-    if (!modal || !this.#modalList.length) {
+    if (!modal || !this.#getTopModal().modal) {
```

Comment:
This is just the same check essentially, why did you change it?

Reply:
Tbh I want to refactor this a little bit to the following, the old implementation won't work as I've updated the list composition so we need to check the modal param against the ref not just the indexed. I could leave it as checking the length of the list but the return at the bottom would potentially evaluate against a topModal that is undefined

```
const { modal: topModal } = this.#getTopModal();
if (!modal || !topModal) {
return false;
}
return modal === topModal;
```